### ISSUE TITLE
invalid validation error

### ISSUE THREAD

**Issue by foolioo** [sirosen/check-jsonschema#25]
I am unsure if this is an issue with check-jsonschema, [jsonschema](https://github.com/Julian/jsonschema/) itself, since the [schema file seems ok to me](https://docs.renovatebot.com/renovate-schema.json).  If you could point what is the source of the issue, I can move the issue there.

Given a [Renovate configuration file](https://docs.renovatebot.com/modules/manager/regex/#configuration-templates) `renovate.json`

```shell
$ cat renovate.json
{
  "regexManagers": [
    {
      "fileMatch": ["^Dockerfile$"],
      "matchStrings": ["ENV YARN_VERSION=(?<currentValue>.*?)\n"],
      "depNameTemplate": "yarn",
      "datasourceTemplate": "npm"
    }
  ]
}
```

When running check-jsonschema it reports an error

```shell
$ check-jsonschema --schemafile https://docs.renovatebot.com/renovate-schema.json renovate.json
Schema validation errors were encountered.
  ./renovate.json::regexManagers.0.matchStrings.0: 'ENV YARN_VERSION=(?<currentValue>.*?)\n' is not a 'regex'
```

But the error does not happen when running jsonschema

```python
import json
import jsonschema

with open("./renovate-schema.json") as f:
  schema = json.load(f)
with open("./renovate.json") as f:
  doc = json.load(f)  
try:
  jsonschema.validate(instance=doc, schema=schema)
except Exception as err:
  print(err)
```

These are the details of my environment

```shell
$ python --version
Python 3.9.9

$ pip list | grep jsonschema
check-jsonschema 0.7.1
jsonschema       3.2.0
```

**Comment by sirosen** [comment]
I believe this is a case of #20 .

In v0.6.0+, check-jsonschema is now enforcing `format` specifications from schemas. The `jsonschema` library does not do so by default, as JSON Schema specifies such validation as non-required.

For now, you can dodge this issue with `--disable-format` to turn off the extra checking.
#20 details some of my thoughts and ideas about how to handle the fact that regex syntaxes vary across languages.

**Comment by foolioo** [comment]
When using `--disable-format`, different error pops

```shell
$ check-jsonschema --schemafile ./renovate-schema.json --disable-format "regex" ./renovate.json
Traceback (most recent call last):
  File "/home/foo/check-jsonschema/venv/bin/check-jsonschema", line 33, in <module>
    sys.exit(load_entry_point('check-jsonschema==0.7.1', 'console_scripts', 'check-jsonschema')())
  File "/home/foo/check-jsonschema/venv/lib/python3.9/site-packages/check_jsonschema-0.7.1-py3.9.egg/check_jsonschema/__init__.py", line 16, in main
  File "/home/foo/check-jsonschema/venv/lib/python3.9/site-packages/check_jsonschema-0.7.1-py3.9.egg/check_jsonschema/checker.py", line 99, in run
  File "/home/foo/check-jsonschema/venv/lib/python3.9/site-packages/check_jsonschema-0.7.1-py3.9.egg/check_jsonschema/checker.py", line 84, in _build_error_map
  File "/home/foo/check-jsonschema/venv/lib/python3.9/site-packages/check_jsonschema-0.7.1-py3.9.egg/check_jsonschema/loaders/instance.py", line 26, in iter_files
  File "/home/foo/check-jsonschema/venv/lib/python3.9/site-packages/identify-1.6.2-py3.9.egg/identify/identify.py", line 41, in tags_from_path
    raise ValueError('{} does not exist.'.format(path))
ValueError: regex does not exist.
```

Could you please elaborate how what is the cause and did I use `--disable-format` as I should?

**Comment by sirosen** [comment]
`--disable-format` doesn't take any arguments. So you've passed `"regex"` as a filename in this case.


The option for `--disable-formats "regex,date"` you may have seen in #20 is something I'm considering, but have not implemented. (Partly because it would be confusing with `--disable-format` as a boolean flag. So it may need a better name.)

**Comment by foolioo** [comment]
Thank you

**Comment by sirosen** [comment]
@foolioo, you may be curious to know that I just released v0.9.0 with `check-renovate` as a built-in hook.
#20 has been resolved with the addition of `--format-regex` as an option, and a default check which ignores certain classes of JS regexes (which cannot be checked in python by the `re` library).

I checked the config in this issue as one of my tests of the behavior, but please let me know with new issues if you try it out and run into problems.
